finetune/imbalance.py

```diff
@@ -0,0 +1,47 @@
+"""
+Utilities for dealing with class imbalance
+"""
+import tensorflow as tf
+import numpy as np
+from collections import Counter
+
+from finetune.target_encoders import LabelEncoder
+from finetune.errors import FinetuneError
+
+
+def class_weight_tensor(class_weights, Y, label_encoder):
+    if class_weights is None:
+        return 
+
+    options = {'linear', 'log', 'sqrt'}
+
+    if isinstance(class_weights, str) and class_weights in options:
+        counts = Counter(Y)
+        max_count = max(counts.values())
+        computed_weights = {}
+        for class_name, count in counts.items():
+            ratio =  max_count / count
+            if class_weights == 'linear':
```

Comment:
Something seems wrong here. Imagining a perfectly balanced set of n classes. ratio becomes n for all classes. Id have thought this would want an /num_classes or something and same for the sqrt scaling. So that in the case of balanced classes, no modification is made.